vertx-web-client/src/test/java/io/vertx/ext/web/client/JsonStreamTest.java

```diff
@@ -0,0 +1,107 @@
+package io.vertx.ext.web.client;
+
+import io.vertx.core.Vertx;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.json.JsonObject;
+import io.vertx.core.parsetools.JsonParser;
+import io.vertx.ext.unit.Async;
+import io.vertx.ext.unit.TestContext;
+import io.vertx.ext.unit.junit.VertxUnitRunner;
+import io.vertx.ext.web.codec.BodyCodec;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Checks the behavior of the {@link io.vertx.ext.web.codec.impl.JsonStreamBodyCodec}.
+ */
+@RunWith(VertxUnitRunner.class)
+public class JsonStreamTest {
+
+  private Vertx vertx;
+  private WebClient client;
+
+  @Before
+  public void setup(TestContext tc) {
+    vertx = Vertx.vertx();
+    client = WebClient.create(vertx, new WebClientOptions().setDefaultPort(8080).setDefaultHost("localhost"));
+    Async async = tc.async();
+
+    vertx.createHttpServer().requestHandler(req -> {
+      int count = Integer.valueOf(req.getParam("count"));
+      String separator = req.getParam("separator");
+      if (separator.equalsIgnoreCase("nl")) {
+        separator = "\n";
+      }
+      if (separator.equalsIgnoreCase("bl")) {
+        separator = "\r\n";
+      }
+      req.response().setChunked(true);
+      for (int i = 0; i < count; i++) {
+        // Send chunks...
+        JsonObject json = new JsonObject().put("count", i).put("data", "some message");
+        Buffer buffer = json.toBuffer();
+        req.response().write(buffer.getBuffer(0, i));
+        req.response().write(buffer.getBuffer(i, buffer.length()));
+        req.response().write(separator);
+      }
+      req.response().end();
+    }).listen(8080, x -> {
```

Comment:
you can use directly : `listen(8080, tc.assertAsyncSuccess())` and get rid of the `Async`, `assertAsyncSuccess()` will prevent the current method to finish in junit until it is resolved (i.e it creates an `Async` instance under the hood).